res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -158,147 +163,123 @@ VCI102.pitch = function(ch, midino, value, status, group) {
-        // select Effect2 of the EffectUnit if shift of the other Deck
-        if (value) {
-            VCI102.fxKnob[ch] = group.slice(0, -1) + "_Effect2]";
-        }
-    } else if (VCI102.fxKnob[ch].slice(-2, -1) > 1) {
-        engine.setValue(VCI102.fxKnob[ch], "next_effect", value > 0);
-    } else {
-        engine.setValue(group, "next_chain", value > 0);
+VCI102.meta = function(ch, value, group, key) {
+    if (VCI102.shift[ch % 2]) {
+        engine.setValue(group, "enabled", value > 0);
     }
+    engine.setValue(group, key, value < 127 ? value / 128 : 1);
 };
 
-VCI102.loopLength = [4, 4, 4, 4];
-
-VCI102.setLoopLength = function(ch, status, value) {
-    var LED = [
-        [0x33, 0x29],
-        [0x29, 0x26],
-        [0x33, 0x29],
-        [0x29, 0x26]
-    ];
-    if (value > 64 || value * 32 < 1) return;
-    VCI102.loopLength[ch] = value;
-    midi.sendShortMsg(status, LED[ch][0], (value < 4) * 127);
-    midi.sendShortMsg(status, LED[ch][1], (value > 4 || value * 4 < 1) * 127);
+VCI102.parameter1 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter1");
+    } else {
+        VCI102.meta(ch, value, VCI102.slot(group, 1), "meta");
+    }
 };
 
-VCI102.loop = function(ch, midino, value, status, group) {
-    if (value) {
-        if (engine.getValue(group, "loop_enabled")) {
-            engine.setValue(group, "reloop_exit", 1);
-        } else {
-            engine.setValue(group, "beatloop", VCI102.loopLength[ch]);
-        }
+VCI102.parameter2 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter2");
+    } else {
+        VCI102.meta(ch, value, VCI102.slot(group, 2), "meta");
     }
 };
 
-VCI102.reloop = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_out", value > 0);
+VCI102.parameter3 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter3");
     } else {
-        engine.setValue(group, "reloop_exit", 1);
+        VCI102.meta(ch, value, VCI102.slot(group, 3), "meta");
     }
 };
 
-VCI102.loop_halve = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_halve", value > 0);
-    } else if (value) {
-        VCI102.setLoopLength(ch, status, VCI102.loopLength[ch] / 2);
+VCI102.parameter4 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter4");
+    } else {
+        VCI102.meta(ch, value, group, "super1");
```

Comment:
> You may want to keep the metaknob of one effect like Filter or Reverb at one position while using the chain and manipulate the metaknobs of other effects. Without the dry/wet knob, you'd have to turn the metaknobs of every effect when mixing the effects in and out, but that does not work well when you want to leave one or more metaknobs in the same position.

For this purpose, we have multiple effect units. For example, using one effect unit for variable effects and another unit for constant effects.

> It also doesn't work well when you are using all 3 effects in a chain, unless you happen to have 3 hands.

"super1" can also be used for changing 3 effects simultaneously.

I think "mix" and "super1" have pros and cons in controlling the whole effect unit. "mix" can control it with ease and acceptable sound quality, but "super1" can control it with fine adjustments (by changing links from prameters) and better sound quality (mixing wet and dry may degrade the sound).